admin_app_2/src/utils/auth.tsx

```diff
@@ -0,0 +1,90 @@
+"use client";
+import { apiCalls } from "@/utils/api";
+import { useRouter, useSearchParams } from "next/navigation";
+import { ReactNode, createContext, useContext, useState } from "react";
+
+type AuthContextType = {
+  token: string | null;
+  user: string | null;
+  accessLevel: "readonly" | "fullaccess";
+  loginError: string | null;
+  login: (username: string, password: string) => void;
+  logout: () => void;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+type AuthProviderProps = {
+  children: ReactNode;
+};
+
+const AuthProvider = ({ children }: AuthProviderProps) => {
+  const [user, setUser] = useState<string | null>(null);
+  const getInitialToken = () => {
+    if (typeof window !== "undefined") {
+      return localStorage.getItem("token");
+    }
+    return null;
+  };
+
+  const [token, setToken] = useState<string | null>(getInitialToken);
+
+  const [loginError, setLoginError] = useState<string | null>(null);
+  const [accessLevel, setAccessLevel] = useState<"readonly" | "fullaccess">(
+    "readonly",
+  );
```

Comment:
IMPORTANT: This default value resets the permission level on page reloads somehow. If I login with fullaccess and refresh a page it goes to readonly mode. If I change this default value to "fullaccess" and login as readonly, reloads reset permissions to fullaccess!

Reply:
Yes - because Auth context is lost on refresh!
I had a few options to fix this:
* log out on refresh - annoying
* save `accessLevel` to localstorage and load from from there. Not safe because user can just edit their local storage to go from `readonly` to `fullaccess`
* save `accessLevel` as an encrypted object with rotating keys. Oof. That's a big change.

I'm open to option 1 or 2. And I think I'll reach out to Griffin or Jay to get ideas on fixing this.

Raised a [ticket](https://idinsight.atlassian.net/browse/AAQ-421) for it.

Reply:
I think the second option sounds like a good compromise right now? It'll behave as the user expects and can be made secure later. Maybe we can make that change with a big warning comment above the code reminding us to change it ASAP (tickets sometimes get lost in the backlog and this seems important).

Reply:
ok! Fixed